src/acp/rule.ts

```diff
@@ -197,3 +211,283 @@ export function setForbiddenRuleOnPolicy(
 export function getForbiddenRuleOnPolicyAll(policy: Policy): UrlString[] {
   return getIriAll(policy, acp.noneOf);
 }
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Initialise a new empty [[SolidDataset]] to store [[Rule]]'s in.
+ */
+export const createRuleDataset = createSolidDataset;
```

Comment:
Thinking: by exposing this function, we are stimulating people to keep their rules in a separate Resource. Would it be better to encourage people to add them to their APRs, i.e. local to the policies that refer to them?

Reply:
I think reuse would be encouraged if the rules were kept in a separate resource: I'd imagine a user wanting to create only one "family" or "dog enthusiasts" rules, and manipulate agents in these, rather than having mostly resource-scoped rules, but I could see how that requires extra trust to the apps you install, if each app may mess up your permissions because they are centralised in a few resources... But even without sharing across apps, for one app having a dedicated resource to list groups with whom data is shared seems easier to manage than having resource-specific groups. What do you think ?

Reply:
I'm not sure if reuse makes sense. Sure, the "family" or "dog enthusiasts" use case makes sense, but that would be covered by a group. A rule, instead, is more for the use case "I want to allow family and disallow dog enthusiasts", which I'm not sure if you really want to re-use those across Policies? Best use case I can think of is "Alice's birthday" being an "allOf <friends>; noneOf <alice>", but even then, I think there'd be _one_ policy called "Alice's birthday", not a rule that's used in multiple policies, and especially not a rule that's used in multiple policies defined in different resources.

Perhaps we should just ask Emmet whether Rules are intended to be reused?

Reply:
Ah wait, no, Rules are even just groups of groups and agents. Just like... Groups. They're just groups specific to ACPs, I guess? I guess maybe then reuse makes sense?

Reply:
Yeah, I'd certainly agree with Nic on the reusability of Rules, although the LucidChart architecture diagram doesn't represent a Rule Resource (just Rule statements). So probably a good idea to get Emmet's input on that question (I'll ping him now).